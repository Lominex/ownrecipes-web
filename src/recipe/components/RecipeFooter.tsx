import { useIntl } from 'react-intl';
import { useSelector } from 'react-redux';

import { CombinedStore } from '../../app/Store';

import MiniBrowse from '../../browse/containers/MiniBrowse';

const RecipeFooter: React.FC = () => {
  const intl = useIntl();

  const miniBrowseState = useSelector((state: CombinedStore) => state.browse.miniBrowse);

  if (!miniBrowseState.hasConnection || miniBrowseState.error) return null;

  return (
    <>
      <hr />
      <article className='recipe-footer'>
        <MiniBrowse heading={intl.messages['nav.home.recommended_recipes_title'] as string} count={4} />
      </article>
    </>
  );
};

export default RecipeFooter;
