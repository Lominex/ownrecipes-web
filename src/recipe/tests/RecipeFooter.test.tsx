import RecipeFooter from '../components/RecipeFooter';
import createComponentWithIntlAndRouter from '../../test/createComponentWithIntlAndRouter';

test('Test Footer without Edit Link', () => {
  const deleteRecipe = jest.fn();
  const component = createComponentWithIntlAndRouter(
    <RecipeFooter
        slug='123'
        source='google.com'
        username='Ryan Noelk'
        updateDate='2017-01-01'
        showEditLink={false}
        deleteRecipe={deleteRecipe}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Test Footer with Edit Link', () => {
  const deleteRecipe = jest.fn();
  const component = createComponentWithIntlAndRouter(
    <RecipeFooter
        slug='123'
        source='google.com'
        username='Ryan Noelk'
        updateDate='2017-01-01'
        showEditLink
        deleteRecipe={deleteRecipe}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
