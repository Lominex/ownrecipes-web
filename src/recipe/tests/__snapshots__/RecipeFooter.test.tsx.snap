// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Footer with Edit Link 1`] = `
<div
  className="card-footer"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        Source: 
        <a
          href="google.com"
        >
          localhost
        </a>
      </div>
      <p
        className="body2"
      >
        Created by: Ryan Noelk
      </p>
      <p
        className="body2"
      >
        Last Updated: 2017-01-01
      </p>
    </div>
    <div
      className="col-auto"
    >
      <div
        className="btn-group"
        role="group"
      >
        <a
          href="/recipe/edit/123"
          onClick={[Function]}
        >
          <button
            className="btn btn-primary btn-sm"
            disabled={false}
            type="button"
          >
            <i
              className="bi bi-pencil-fill"
            />
          </button>
        </a>
        <button
          className="btn btn-danger btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <i
            className="bi bi-trash-fill size-1x"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test Footer without Edit Link 1`] = `
<div
  className="card-footer"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        Source: 
        <a
          href="google.com"
        >
          localhost
        </a>
      </div>
      <p
        className="body2"
      >
        Created by: Ryan Noelk
      </p>
      <p
        className="body2"
      >
        Last Updated: 2017-01-01
      </p>
    </div>
  </div>
</div>
`;
